src/bin/version/main.rs

```diff
@@ -0,0 +1,101 @@
+extern crate core;
+
+use colored::Colorize;
+use std::ffi::OsString;
+use std::path::Path;
+use std::{env, fs};
+
+const SDKMAN_DIR_ENV_VAR: &str = "SDKMAN_DIR";
+const DEFAULT_SDKMAN_HOME: &str = ".sdkman";
+const VERSION_FILE: &str = "var/version";
+
+fn main() {
+    let sdkman_dir = infer_sdkman_dir();
+    let version = locate_version_file(sdkman_dir).and_then(read_content);
+    match version {
+        Some(content) => println!("\n{} {}", "SDKMAN".yellow(), content.yellow()),
+        None => std::process::exit(exitcode::CONFIG),
+    }
+}
+
+fn infer_sdkman_dir() -> String {
+    match env::var(SDKMAN_DIR_ENV_VAR) {
+        Ok(s) => s,
+        Err(_) => fallback_sdkman_dir(),
+    }
+}
+
+fn fallback_sdkman_dir() -> String {
+    dirs::home_dir()
+        .map(|dir| dir.join(DEFAULT_SDKMAN_HOME))
```

Comment:
If I remember correctly using `.join` on a Path in Rust was relatively expensive. Probably better to use `PathBuf` to construct `Path`s.

`dirs::home_dir` already returns a `PathBuf` btw 😄 .

Reply:
That's right, I'm mapping over the `Option` that `dirs::home_dir()` returns and joining `.sdkman` onto the `PathBuf`.

Reply:
All cut over to `PathBuf` now.